minimum_jerk_ros/src/minimum_jerk_ros.cpp

```diff
@@ -0,0 +1,736 @@
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_activate(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_activate()...");
+        this->action_rotation_server_->activate();
+        this->action_translation_server_->activate();
+        this->publisher_->on_activate();
+        this->obstacles_marker_pub_->on_activate();
+        this->goal_pose_pub_->on_activate();
+        this->trajectory_pub_->on_activate();
+        RCLCPP_INFO(this->get_logger(), "Action server successfully activated...");
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_deactivate(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_deactivate()...");
+        this->action_rotation_server_->deactivate();
+        this->action_translation_server_->deactivate();
+        this->publisher_->on_deactivate();
+        this->obstacles_marker_pub_->on_deactivate();
+        this->goal_pose_pub_->on_deactivate();
+        this->trajectory_pub_->on_deactivate();
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_cleanup(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_cleanup()...");
+        this->publisher_.reset();
+        this->buffer_.reset();
+        this->listener_.reset();
+        this->collision_srv_client_.reset();
+        this->collision_client_node_.reset();
+        this->obstacles_marker_pub_.reset();
+        this->goal_pose_pub_.reset();
+        this->trajectory_pub_.reset();
+
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_shutdown(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_shutdown()...");
+        this->publisher_.reset();
+        this->goal_pose_pub_.reset();
+        this->trajectory_pub_.reset();
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    void MinimumJerkRos::rotation_callback()
+    {
+        auto goal = action_rotation_server_->get_current_goal();
+        RCLCPP_INFO(this->get_logger(), "Rotation: %f rad, min_vel: %f, collision_check: %i, yaw_goal_tolerance: %f, data_save: %i",
+                    goal->target_yaw, goal->min_rotational_vel, goal->enable_collision_check, goal->yaw_goal_tolerance, goal->enable_data_save);
+
+        if (goal->target_yaw < -M_PI || goal->target_yaw > M_PI)
```

Comment:
Wonder what would happen in the library if you pass `target_yaw = 10.0`

Reply:
this is possible with the library